meteor/client/ui/Shelf/TimelineDashboardPanel.tsx

```diff
@@ -88,6 +88,7 @@ export const TimelineDashboardPanel = translateWithTracker<
 		render() {
 			if (this.props.visible && this.props.showStyleBase && this.props.filter) {
 				const filter = this.props.filter as DashboardLayoutFilter
+				const uniquenessIds = new Set<string>()
```

Comment:
There's a lot of this assigning an empty set and then passing it into a function. Seems unnecessary? Why not just pass a `new Set()` into the filtering function? Why not just have the filtering function build itself a new Set, if none is provided?

Reply:
Because `matchFilter` is run in a loop where each call operates on a single item. Also, in e.g. the `AdLibPanel` we are passing the same set to the `renderRundownAdLibs` and `renderSegments` functions to handle the case where two equivalent adlibs are created in both the baseline and a segment, so we need a set which lives long enough.

Reply:
Ha, yeah, you're totally correct! I completely missed the fact that this needs to be consistent across the entire filter invocation. Sorry!